includes/Sync/Product_Import.php

```diff
@@ -807,14 +807,31 @@ protected function extract_square_item_variation_data( $variation ) {
 	 *
 	 * @since 4.9.0
 	 *
-	 * @param array $data the product data
-	 * @return int
+	 * @param array $options    the Square options
+	 * @param array $variations the product variations to determine which option values are actually used
+	 * @return array
 	 * @throws \Exception
 	 */
-	protected function extract_attributes_from_square_options( $options ) {
+	protected function extract_attributes_from_square_options( $options, $variations = array() ) {
 
 		$data_attributes = array();
 
+		// Collect all option values that are actually used by variations.
+		$used_option_values = array();
+		foreach ( $variations as $variation ) {
+			if ( isset( $variation['attributes'] ) ) {
```

Comment:
Are we confident this will always be an array? May be worth being a bit defensive here
```suggestion
			if ( isset( $variation['attributes'] ) && is_array( $variation['attributes'] ) ) {
```

Reply:
This will always include an `attributes` item in the array, though it’s fine to be extra defensive.